File: web/src/lib/server/utils/validation/playlist-track-addition-schema.ts
import { z } from "zod";

export const playlistTrackAdditionSchema = z.object({
  author: z.string(),
  thumbnail: z.string(),
  duration: z.string(),
  isExplicit: z.boolean(),
  title: z.string().max(255),
  videoId: z.string().max(11),
  videoType: z.enum(["track", "uvideo"]),
});

File: server/models/schemas/playlist_track_addition_schema.go
package schemas

import type_constants "wavelength/constants/types"

type PlaylistTrackAdditionSchema struct {
	Author     string                           `json:"author"`
	Thumbnail  string                           `json:"thumbnail"`
	Duration   string                           `json:"duration"`
	IsExplicit bool                             `json:"isExplicit"`
	Title      string                           `json:"title"`
	VideoId    string                           `json:"videoId"`
	VideoType  type_constants.PlaylistTrackType `json:"videoType"`
}
